### PR TITLE
[@types/react-native] Add setNativeProps to ScrollView

### PR DESCRIPTION
**Pull request by abmantis**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://reactnative.dev/docs/direct-manipulation
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -6790,6 +6790,14 @@ export class ScrollView extends ScrollViewBase {
...
      */
     scrollWithoutAnimationTo?: (y: number, x: number) => void;
+    
+    /**
+     * This function sends props straight to native. They will not participate in
+     * future diff process - this means that if you do not include them in the
+     * next render, they will remain active (see [Direct
+     * Manipulation](docs/direct-manipulation.html)).
+     */
+    setNativeProps(nativeProps: Object): void;
```
Comment (ca057) [inline types/react-native/index.d.ts:6800]: Shouldn’t this be the primitive `object`?
```suggestion
    setNativeProps(nativeProps: object): void;
```

In addition, I think we should change the link to the docs to not be relative but absolute, as the types are coming from a different package than the React Native docs :)

Do you have a link to the source code where this function is defined? Aaand can we add a test for this? :D

**Comment by abmantis** [comment]
This is not abandoned! :D I will update the PR with the suggestions ASAP.

File: types/react-native/index.d.ts (hunk 1 above)
Comment (abmantis) [reply types/react-native/index.d.ts:6820]: This was mostly copied from the other `setNativeProps` method in NativeMethods. I'll add your suggestions. Regarding the test, I'll try my best :D

**Comment by abmantis** [comment]
@ca057 Done. The CI is failing, but I've not touched animated.tsx...

**Review by ca057 — APPROVED** [review]

**Comment by alloy** [comment]
@abmantis The CI issue should be fixed on master, can you rebase your changes?

**Review by ca057 — APPROVED** [review]

**Review by a-tarasyuk — APPROVED** [review]

**Comment by alloy** [comment]
Ace, thanks! 🙏

**Pull request merged**